### PR TITLE
feat(tfhe): replace asm with rust intrinsics

### PR DESCRIPTION
**Pull request by sarah-ek**
closes: _please link all relevant issues_

### Check-list:

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [x] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

File: tfhe/src/core_crypto/fft_impl/fft64/math/fft/x86.rs [hunk 1]
```diff
@@ -89,52 +89,10 @@ pub fn mm256_cvtpd_epi64(simd: V3, x: __m256d) -> __m256i {
...
-
-        as_i64x8
-    }
-    let _ = simd.avx512dq;
-
     // SAFETY: simd contains an instance of avx512dq, that matches the target feature of
     // `implementation`
-    unsafe { implementation(x) }
+    _ = simd;
+    unsafe { _mm512_cvttpd_epi64(x) }
```
Comment (IceTDrinker) [inline tfhe/src/core_crypto/fft_impl/fft64/math/fft/x86.rs:95]: is this going to get inlined if we don't have the target feature on the function itself ?

also could we maybe replace our function call by this intrinsics directly ?

last question, I think you said you had an improvement in performance by changing the conversion function ? here it seems it's the same ?

File: tfhe/src/core_crypto/fft_impl/fft64/math/fft/x86.rs [hunk 2]
```diff
@@ -174,52 +132,10 @@ pub fn mm256_cvtepi64_pd(simd: V3, x: __m256i) -> __m256d {
...
-
-        as_f64x8
-    }
-    let _ = simd.avx512dq;
-
     // SAFETY: simd contains an instance of avx512dq, that matches the target feature of
     // `implementation`
-    unsafe { implementation(x) }
+    _ = simd;
+    unsafe { _mm512_cvtepi64_pd(x) }
```
Comment (IceTDrinker) [inline tfhe/src/core_crypto/fft_impl/fft64/math/fft/x86.rs:138]: same questions here :)

**Review by IceTDrinker — COMMENTED** [review]
Thanks, just a few questions, I might be misremembering some things

File: tfhe/src/core_crypto/fft_impl/fft64/math/fft/x86.rs (hunk 1 above)
Comment (sarah-ek) [reply tfhe/src/core_crypto/fft_impl/fft64/math/fft/x86.rs:95]: should get inlined if it works like the other intrinsics.
i also figured I'd leave the instruction change for a separate pr

Comment (IceTDrinker) [reply tfhe/src/core_crypto/fft_impl/fft64/math/fft/x86.rs:95]: sounds good then !

**Review by IceTDrinker — APPROVED** [review]
Thanks a lot :)

**Pull request merged**